### PR TITLE
fix: Fixing a series of javadoc warnings

### PR DESCRIPTION
**Pull request by hiranya911**
Fixing the following javadoc warnings:

```
[WARNING] Javadoc Warnings
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:798: warning 101: Unresolved link/see tag "IllegalArgumentException" in com.google.firebase.auth.AbstractFirebaseAuth
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:820: warning 101: Unresolved link/see tag "IllegalArgumentException" in com.google.firebase.auth.AbstractFirebaseAuth
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:873: warning 101: Unresolved link/see tag "IllegalArgumentException" in com.google.firebase.auth.AbstractFirebaseAuth
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:889: warning 103: Unknown tag: @throw
[WARNING] ~hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:905: warning 103: Unknown tag: @throw
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/ListProviderConfigsPage.java:104: warning 101: Unresolved link/see tag "Iterable" in com.google.firebase.auth.ListProviderConfigsPage
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/ListProviderConfigsPage.java:104: warning 101: Unresolved link/see tag "Iterator" in com.google.firebase.auth.ListProviderConfigsPage
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/ListProviderConfigsPage.java:107: warning 101: Unresolved link/see tag "Iterable" in com.google.firebase.auth.ListProviderConfigsPage
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/ListProviderConfigsPage.java:118: warning 101: Unresolved link/see tag "Iterable" in com.google.firebase.auth.ListProviderConfigsPage
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/ListProviderConfigsPage.java:120: warning 101: Unresolved link/see tag "Iterable" in com.google.firebase.auth.ListProviderConfigsPage
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/OidcProviderConfig.java:131: warning 101: Unresolved link/see tag "AbstractFirebaseAuth#updateOidcProviderConfig(CreateRequest)" in com.google.firebase.auth.OidcProviderConfig.UpdateRequest
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/multitenancy/ListTenantsPage.java:101: warning 101: Unresolved link/see tag "Iterable" in com.google.firebase.auth.multitenancy.ListTenantsPage
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/multitenancy/ListTenantsPage.java:101: warning 101: Unresolved link/see tag "Iterator" in com.google.firebase.auth.multitenancy.ListTenantsPage
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/multitenancy/ListTenantsPage.java:104: warning 101: Unresolved link/see tag "Iterable" in com.google.firebase.auth.multitenancy.ListTenantsPage
[WARNING] ~/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/multitenancy/ListTenantsPage.java:115: warning 101: Unresolved link/see tag "Iterable" in com.google.firebase.auth.multitenancy.ListTenantsPage
[WARNING] ~/hkj/Projects/firebase-admin-java/public/src/main/java/com/google/firebase/auth/multitenancy/ListTenantsPage.java:117: warning 101: Unresolved link/see tag "Iterable" in com.google.firebase.auth.multitenancy.ListTenantsPage
```

Specifically we cannot `{@link }` to classes that are not part of our codebase.

### CODE REVIEW

File: src/main/java/com/google/firebase/auth/OidcProviderConfig.java [hunk 1]
```diff
@@ -132,7 +132,7 @@ CreateRequest getThis() {
      * provider.
      *
      * <p>The returned object should be passed to
-     * {@link AbstractFirebaseAuth#updateOidcProviderConfig(CreateRequest)} to save the updated
+     * {@link AbstractFirebaseAuth#updateOidcProviderConfig(UpdateRequest)} to save the updated
```
Comment (lahirumaramba) [inline src/main/java/com/google/firebase/auth/OidcProviderConfig.java:135]: Was this a bug? :)

File: src/main/java/com/google/firebase/auth/ListProviderConfigsPage.java [hunk 2]
```diff
@@ -261,7 +261,7 @@ public ListSamlProviderConfigsResponse fetch(int maxResults, String pageToken)
 
     ListProviderConfigsPage<T> create() throws FirebaseAuthException {
       ListProviderConfigsResponse<T> batch = source.fetch(maxResults, pageToken);
-      return new ListProviderConfigsPage<T>(batch, source, maxResults);
+      return new ListProviderConfigsPage<>(batch, source, maxResults);
```
Comment (lahirumaramba) [inline src/main/java/com/google/firebase/auth/ListProviderConfigsPage.java:264]: Should we do a project wide search for similar missing generics? :)

File: src/main/java/com/google/firebase/auth/OidcProviderConfig.java (hunk 1 above)
Comment (hiranya911) [reply src/main/java/com/google/firebase/auth/OidcProviderConfig.java:135]: Yes it was.

File: src/main/java/com/google/firebase/auth/ListProviderConfigsPage.java (hunk 2 above)
Comment (hiranya911) [reply src/main/java/com/google/firebase/auth/ListProviderConfigsPage.java:264]: We should at some point. Hopefully we can get an IDE to do the work all in one go.

**Pull request merged**